redash/query_runner/json_ds.py

```diff
@@ -5,10 +5,15 @@
 import datetime
 from urlparse import urlparse
 from funcy import compact, project
+from six import text_type
 from redash.utils import json_dumps
 from redash.query_runner import (BaseHTTPQueryRunner, register,
                                  TYPE_BOOLEAN, TYPE_DATETIME, TYPE_FLOAT,
                                  TYPE_INTEGER, TYPE_STRING)
+try:
+    long
+except NameError:
+    long = int
```

Comment:
I would suggest to add a `redash/utils/compat.py` file for this stuff.

Reply:
Given that the only place that __long__ appears is in types_maps in
* redash/query_runner/couchbase.py
* redash/query_runner/elasticsearch.py
* redash/query_runner/json_ds.py
* redash/query_runner/mongodb.py

Perhaps it would be a better choice to create a single types_map that these query runners can import.

Reply:
Given that __couchbase.py__ is conflicted about what datatype a __datetime.datetime__ is, I just followed @jezdez recommended solution above.

Reply:
Yeah, let's not overthink this, this will be code to be removed when Redash moves to Python 3 after all.